File: Part 2 - Regression/Section 4 - Simple Linear Regression/simple_linear_regression.R
# Data preprocessing
dataset = read.csv('Salary_Data.csv')
#dataset = dataset[, 2:3]

# Splitting the data into the training and test data sets
#install.packages('caTools')
set.seed(123)
split = sample.split(dataset$YearsExperience, SplitRatio = 2/3)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

# Feature scaling
# training_set[, 2:3] = scale(training_set[, 2:3])
# test_set[, 2:3] = scale(test_set[, 2:3])

File: Part 2 - Regression/Section 4 - Simple Linear Regression/simple_linear_regression.py
# -*- coding: utf-8 -*-

# Simple linear regression

#Importing libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
"""
Created on Mon Nov 20 22:03:44 2017

@author: janne_000
"""
dataset = pd.read_csv('Salary_Data.csv')
# take all values except last column
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 1].values


# Splitting the data into the training and test data sets
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 1/3, random_state = 0)

# Feature scaling
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)"""
